recipes_source/zero_redundancy_optimizer.rst

```diff
@@ -0,0 +1,154 @@
+Shard Optimizer States with ZeroRedundancyOptimizer
+===================================================
+
+.. note:
+    `ZeroRedundancyOptimizer` is introduced in PyTorch 1.8 as a prototype
+    feature. It API is subject to change.
+
+In this recipe, you will learn:
+
+- The high-level idea of `ZeroRedundancyOptimizer`.
+- How to use `ZeroRedundancyOptimizer` in distributed training and its impact.
+
+
+Requirements
+------------
+
+- PyTorch 1.8+
+- `Getting Started With Distributed Data Parallel <https://pytorch.org/tutorials/intermediate/ddp_tutorial.html>`_
+
+
+What is `ZeroRedundancyOptimizer`?
+----------------------------------
+
+The idea of `ZeroRedundancyOptimizer` comes from the
+`DeepSpeed/ZeRO project <https://github.com/microsoft/DeepSpeed>`_ that shards
```

Comment:
we can add another reference here, I can look that up, the idea is older than that actually and it was in a public MLP framework

Reply:
It has been in Marian for a while it seems, before Zero 
https://github.com/marian-nmt/marian-dev

https://github.com/marian-nmt/marian-dev/blob/737f43014a939a3ded2806b00cbaa661fbcc5f49/src/training/communicator_nccl.h#L228

Reply:
added